### ISSUE TITLE
Module path not updated to github.com/gorilla/websocket/v2 and new tag does not follow semantic versioning

### ISSUE THREAD

**Issue by FZambia** [gorilla/websocket#860]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

In yesterday's [v2 release](https://github.com/gorilla/websocket/releases/tag/v2.0) I noticed two issues:

1. Path should be updated in go.mod. See https://go.dev/blog/v2-go-modules and specifically:

```
If an old package and a new package have the same import path,
the new package must be backwards compatible with the old package.
```

2. New v2.0 tag does not follow semantic versioning - please take a look how all the previous tags were named.

### Expected Behavior

When releasing v2 version of module it's recommended to also update module path. So I suppose it must be:

```
github.com/gorilla/websocket/v2
```

And tag is v2.0.0

### Steps To Reproduce

https://github.com/gorilla/websocket/blob/ac0789be11725ab2285233e9a3800c2312cff4fc/go.mod#L1

https://github.com/gorilla/websocket/tree/v2.0

### Anything else?

I suppose this may cause issues with Go tooling.

Also, could you please tell why there is `vendor` directory appeared in repo root? It's quite strange to see it in 2023.

**Comment by FZambia** [comment]
Never used it myself, but probably may be useful - [retract](https://go.dev/ref/mod#go-mod-file-retract) directive of go.mod

**Comment by coreydaley** [comment]
This should be fixed now, release of gorilla/websocket is v1.5.1
